crates/gen-rust-wasm/src/lib.rs

```diff
@@ -941,10 +980,24 @@ impl Bindgen for FunctionBindgen<'_> {
         format!("ptr{}", tmp)
     }
 
-    fn i64_return_pointer_area(&mut self, amt: usize) -> String {
+    fn i64_return_pointer_area(&mut self, iface: &Interface, export: bool, amt: usize) -> String {
         assert!(amt <= self.gen.i64_return_pointer_area_size);
         let tmp = self.tmp();
-        self.push_str(&format!("let ptr{} = RET_AREA.as_mut_ptr() as i32;\n", tmp));
+        // For exports in standalone mode, the `RET_AREA` is accessed via a macro from another
+        // crate, so we need to access it from the expected crate name.
```

Comment:
I haven't run this to see the generated code so this might be a silly question, but do we need to use a separate `RET_AREA` for each use of the `export!()` macro? 

My interpretation based on the PR's discussion is that the generated `export!()` macro will expand to the code that gets generated by `wit_bindgen_rust::export!()` at the moment, minus the struct and trait definitions. Which means the `RET_AREA` static will be accessible in the current scope and generated functions can refer to it freely.

It might be easier to look at some code:

```rust
// wit_bindgen_rust::export!("export.wit") expands to something like this:

pub mod export {
    pub trait Export {
        fn say_something() -> String;
    }

    /// Given the name of something implementing the [`Export`] trait, generate
    /// glue code that uses the type's functionality to implement the `export.wit`
    /// interface.
    macro_rules! export {
        ($t:ty) => {
            mod export {
                use super::Export; // Assume the "Export" trait was imported in the parent namespace

                #[export_name = "export-0.1.0#say-something"]
                unsafe extern "C" fn __wit_bindgen_export_say_something() -> i32{
                    ...

                    let result0 = <$t as Export>::say_something();

                    ...

                    let ptr2 = RET_AREA.as_mut_ptr() as i32;

                    ...
                    ptr2
                }

                static RET_AREA: [...] = ...;
            }
        }
    }
}
```

Downstream users might then use it like so:

```rust
use common_interface_crate::export::Export;

struct ExportImpl;

impl Export for ExportImpl {
  ...
}

common_interface_crate::export::export!(ExportImpl);
```

Or if we don't want to assume the `Export` trait is available in the parent namespace we could change the `export!()` macro's syntax to something like `export!(ExportImpl as Export)`.

Reply:
`RET_AREA` currently can't be shared between interfaces as it's calculated to be a maximum space needed between all referencing functions from a single interface (each `export!` invocation can potentially know only about a single interface at a time).

The generated `export!` from these changes could definitely move the `RET_AREA` into the macro so it can just live static next to the functions and inaccessible from the outside world, at least. I can definitely make that change.